integration/combination/test_api_with_gateway_responses.py

```diff
@@ -1,9 +1,12 @@
+import logging
 from unittest.case import skipIf
 
+from tenacity import stop_after_attempt, retry_if_exception_type, after_log, wait_exponential, retry, wait_random
```

Comment:
Where is tenacity coming from?

Reply:
[Tenacity](https://pypi.org/project/tenacity/) is a retry library we used in integration tests and it has Apache 2.0 liciense.